src/util/dstring.h

```diff
@@ -25,11 +27,11 @@ class dstringt
   }
 
   // this is safe for static objects
-  // the 2nd argument is to avoid accidental conversions
+  // marked explicit to avoid accidental conversions
   #ifdef __GNUC__
   constexpr
   #endif
-  dstringt(unsigned _no, unsigned):no(_no)
+  explicit dstringt(unsigned _no):no(_no)
   {
   }
 
```

Comment:
In that case, I'd be inclined to make this constructor private (but still 1-param explicit), and have a `friend dstringt make_dstring_using_table_index(unsigned)` (or similar) which calls the private constructor. I think having the dummy parameter makes sense at the definition (when commented), but is confusing at the call site. Having a separate function is more self-documenting.